engine/src/main/java/org/camunda/bpm/engine/impl/interceptor/CommandCounterInterceptor.java

```diff
@@ -30,15 +30,14 @@ public CommandCounterInterceptor(ProcessEngineConfigurationImpl processEngineCon
 
   @Override
   public <T> T execute(Command<T> command) {
-
     try {
       return next.execute(command);
     } finally {
       TelemetryRegistry telemetryRegistry = processEngineConfiguration.getTelemetryRegistry();
       if (telemetryRegistry != null) {
         String className = ClassNameUtil.getClassNameWithoutPackage(command);
         // anonymous class/lambda implementations of the Command interface are excluded
-        if (!command.getClass().isAnonymousClass() && !className.contains("$$Lambda$")) {
+        if (!command.getClass().isAnonymousClass() && !ClassNameUtil.isLambda(className)) {
```

Comment:
❌ As discussed, we favour more strict check for all JDK versions (implying less lines of code at the end):
```suggestion
        if (!command.getClass().isAnonymousClass() && !className.contains("$$Lambda")) {
```

Reply:
Related to the initial [solution](https://github.com/camunda/camunda-bpm-platform/issues/4077#:~:text=Solution%3A%20Apply%20the%20new%20check%20only%20for%20Java21%20and%20keep%20the%20legacy%20behaviour%20for%20JDKs%20%3C%2021%20(conservative%20approach%20instead%20of%20checking%20only%20for%20%22%24%24Lambda%22%20suffixes)) for the different JDK lambda name notation